lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1921,19 +1934,51 @@ def _get_deployable_vhd_path(
                     if self._is_stuck_copying(blob_client, log):
                         # Delete the stuck vhd.
                         blob_client.delete_blob(delete_snapshots="include")
-                    elif original_key == cached_key:
-                        # If it exists, return the link, not to copy again.
-                        log.debug("the sas url is copied already, use it directly.")
-                        vhd_exists = True
+                    elif original_key and cached_key:
+                        if original_key == cached_key:
+                            log.debug("the sas url is copied already, use it directly.")
+                            vhd_exists = True
+                        else:
+                            log.debug("found cached vhd, but the hash key mismatched.")
                     else:
-                        log.debug("found cached vhd, but the hash key mismatched.")
+                        log.debug(
+                            "No md5 content either in original blob or current blob. "
+                            "Then no need to check the hash key"
+                        )
+                        vhd_exists = True
 
             if not vhd_exists:
-                blob_client.start_copy_from_url(
-                    original_vhd_path, metadata=None, incremental_copy=False
-                )
+                azcopy_path = self._azure_runbook.azcopy_path
+                log.debug(f"AzCopy path:{azcopy_path}")
+                if azcopy_path and os.path.exists(azcopy_path):
```

Comment:
If the path specified, but not exists, it should raise an exception to make it obvious.

Reply:
Done